language/move-vm/runtime/src/loader.rs

```diff
@@ -893,7 +893,7 @@ impl Loader {
     pub(crate) fn load_type(
         &self,
         type_tag: &TypeTag,
-        data_store: &impl DataStore,
+        data_store: &dyn DataStore,
```

Comment:
@wrwg or @vgao1996, thoughts here? 

I don't really care just wanted to flag

Reply:
That is not good since all calls now go over dynamic dispatch. Why is this needed?

Reply:
Yeah this shouldn't be needed. You can pass a `&impl Trait` to `&dyn Trait`

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=59810b6cd41987904ba479800a708871

Reply:
Regarding why the switch from `impl` to `dyn` for all those calls, it's because the `DataStore` in `NativeContext` is a  `&'a mut dyn DataStore` which cannot be passed as an `&impl DataStore` because its size is unknown (AFAIU).

I therefore changed all the `&impl`'s to `&dyn`'s.

I did not try changing the `&'a mut dyn DataStore` to a `&'a mut impl DataStore` though... maybe that would've worked.

@vgao1996, yes you can do that but I needed to do the opposite.